pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/AggregationGroupByTrimmingService.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.operator.aggregation.groupby;
+
+import com.linkedin.pinot.common.response.broker.GroupByResult;
+import com.linkedin.pinot.core.operator.aggregation.AggregationFunctionContext;
+import com.linkedin.pinot.core.operator.aggregation.function.AggregationFunction;
+import com.linkedin.pinot.core.operator.aggregation.function.AggregationFunctionFactory;
+import com.linkedin.pinot.core.operator.aggregation.function.AggregationFunctionUtils;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import javax.annotation.Nonnull;
+
+
+/**
+ * The <code>AggregationGroupByTrimmingService</code> class provides trimming service for aggregation group-by query.
+ */
+// TODO: revisit the trim algorithm, implement trim on all Object but not only for Comparable.
+public class AggregationGroupByTrimmingService {
+  public static final String GROUP_KEY_DELIMITER = "\t";
+
+  private final int _numAggregationFunctions;
+  private final boolean[] _minOrders;
+  private final int _groupByTopN;
+  private final int _trimSize;
+  private final int _trimThreshold;
+
+  public AggregationGroupByTrimmingService(@Nonnull AggregationFunctionContext[] aggregationFunctionContexts,
+      int groupByTopN) {
+    _numAggregationFunctions = aggregationFunctionContexts.length;
+    _minOrders = new boolean[_numAggregationFunctions];
+    for (int i = 0; i < _numAggregationFunctions; i++) {
+      String aggregationFunctionName = aggregationFunctionContexts[i].getAggregationFunction().getName();
+      if (aggregationFunctionName.equals(AggregationFunctionFactory.AggregationFunctionType.MIN.getName())
+          || aggregationFunctionName.equals(AggregationFunctionFactory.AggregationFunctionType.MINMV.getName())) {
+        _minOrders[i] = true;
+      }
+    }
+    _groupByTopN = groupByTopN;
```

Comment:
Please add the comments explaining the algorithm to derive trimSize and trimThreshold.